### DISCUSSION TITLE
How to work with elements that have the same name and different attributes

### DISCUSSION THREAD

**Discussion by GpixelProg** [pdvrieze/xmlutil#171] [Q&A]
Hello Team,

I have an XML file of this type:

```
<obj id="1">
     <name/>
     <color>white</color>
     <anim>sample0</anim>
     <user id="1"/>
     <user id="2"/>
     <hold_up_state/>
     <main_action id="1"/>
     <main_action id="2"/>
</obj>
```
How can I describe a Data class to work with these elements? - user id="1", ' user id="2"

Regards, 
Maksym

**Comment by pdvrieze** [comment]
In this case it seems that you just have a `User` type with an `id` attribute. However, looking at this data, it seems that you may need to have a transformation to go from the serialization to a semantic type (a good way is to use a private serialization delegate and a custom serializer that delegates to the delegate's serializer.)
